presto-raptor/src/main/java/com/facebook/presto/raptor/storage/organization/Tuple.java

```diff
@@ -27,19 +29,21 @@
         implements Comparable<Tuple>
 {
     private final List<Type> types;
-    private final List<Object> values;
+    private final List<Optional<Object>> values;
 
-    public Tuple(Type type, Object value)
+    public static Tuple of(List<Type> types, Object... values)
     {
-        this(ImmutableList.of(type), ImmutableList.of(value));
+        return new Tuple(types, ImmutableList.copyOf(values).stream()
+                .map(Optional::ofNullable)
+                .collect(Collectors.toList()));
     }
 
-    public Tuple(List<Type> types, Object... values)
+    public Tuple(Type type, Optional<Object> value)
     {
-        this(types, ImmutableList.copyOf(values));
+        this(ImmutableList.of(type), ImmutableList.of(requireNonNull(value, "value is null")));
```

Comment:
The `requireNonNull` check is not needed since `ImmutableList` does not allow nulls